### PR TITLE
Google PubSub native support

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc [hunk 1]
```diff
@@ -32,3 +32,12 @@ Please refer to the above link for usage and configuration details.
 ----
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Camel Quarkus limitations
+
+By default, the Camel PubSub component uses JDK object serialization via `ObjectOutputStream` whenever the message body is anything other than `String` or `byte[]`.
+
+Since such serialization is not yet supported by GraalVM, this extension provides a custom Jackson based serializer to serialize complex message payloads as JSON.
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc:40]: The custom Jackson based serializer is used in both JVM and native modes (the text seems to suggest that).

BTW, we'll see by tomorrow, whether serialization will be supported in GraalVM 21 https://github.com/oracle/graal/pull/2730#issuecomment-737135245

Comment (jamesnetherton) [reply docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc:40]: > The custom Jackson based serializer is used in both JVM and native modes

Yes, it's better IMO to be consistent and have the same behaviour in both modes.

File: docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc [hunk 2]
```diff
@@ -32,3 +32,12 @@ Please refer to the above link for usage and configuration details.
...
+
+If your payload contains binary data, then you will need to handle that by creating a custom Jackson Serializer / Deserializer.
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc:42]: I guess the readers could find this link to Quarkus Jackson guide helpful https://quarkus.io/guides/writing-extensions#customizing-jackson where they explain how to write serializers.

**Review by ppalaga — APPROVED** [review]
Looks good. The docs can be improved in a follow up.

**Pull request merged**